napari/_qt/_tests/test_qt_viewer.py

```diff
@@ -441,3 +443,47 @@ def on_click(layer, event):
 
     viewer.dims.ndisplay = 3
     view._process_mouse_event(mouse_press_callbacks, mouse_event)
+
+
+def test_memory_leaking(make_napari_viewer, tmp_path):
+    data = np.zeros((5, 20, 20, 20), dtype=int)
+    data[1, 0:10, 0:10, 0:10] = 1
+    viewer = make_napari_viewer()
+    image = weakref.ref(
+        viewer.add_image(data, scale=(1, 2, 1, 1), translate=(5, 5, 5))
+    )
+    labels = weakref.ref(
+        viewer.add_labels(data, scale=(1, 2, 1, 1), translate=(5, 5, 5))
+    )
+    del viewer.layers[0]
+    del viewer.layers[0]
+    gc.collect()
+    gc.collect()
+    assert image() is None
+    assert labels() is None
+
+
+def test_leaks_image(make_napari_viewer):
```

Comment:
can you decorate this and the labels test to avoid local popups?

```
from napari._tests.utils import skip_local_popups

@skip_local_popups
```

Reply:
Why other tests in this file does not need this?

Reply:
because they don't use `make_napari_viewer(show=True)`

Reply:
>  (but i think adding `show` is probably a good idea, in case there are only references that are created when visible objects are created?)

Even if now there is no such reference it is possible that it could be added in the future and it will be nice for our tests to catch such leaking problem.